buildSrc/src/main/kotlin/SqlDelightTmdbCacheDefinitionsGenerator.kt

```diff
@@ -0,0 +1,110 @@
+    private val languageColumn = SqlColumn.text("language", "io.github.couchtracker.tmdb.TmdbLanguage")
+
+    /** Which caches to create */
+    private val caches = listOf(
+        SqlTable(
+            name = "MovieDetailsCache",
+            key = listOf(movieIdColumn, languageColumn),
+            value = SqlColumn.text("details", "app.moviebase.tmdb.model.TmdbMovieDetail"),
+        ),
+        SqlTable(
+            name = "MovieReleaseDatesCache",
+            key = listOf(movieIdColumn),
+            value = SqlColumn.text("releaseDates", "kotlin.collections.List<app.moviebase.tmdb.model.TmdbReleaseDates>"),
+        ),
+    )
+
+    /** Which sql functions to generate */
+    private val sqlFunctions = listOf(
+        SqlFunction("get") { cache ->
+            append("SELECT ${cache.value.name}\n")
+            append("FROM ${cache.name}\n")
+            append("WHERE " + cache.key.joinToString(" AND ") { "${it.name} = ?" } + ";\n")
+        },
+        SqlFunction("put") { cache ->
+            append("INSERT INTO ${cache.name}\n")
+            append("VALUES (${cache.key.joinToString { ":${it.name}" }}, :${cache.value.name})\n")
+            append("ON CONFLICT DO UPDATE SET ${cache.value.name} = :${cache.value.name};\n")
+        },
+    )
+
+    init {
+        group = "io.github.couchtracker"
+        description = "Generates the SqlDelight definitions for local the Tmdb Cache"
+    }
+
+    private data class SqlColumn(
+        val name: String,
+        val sqlType: String,
+        val kotlinType: String,
+        val nullable: Boolean = false,
+    ) {
+        fun columnDefinition() = buildString {
+            append("$name $sqlType AS $kotlinType")
+            if (!nullable) {
+                append(" NOT NULL")
+            }
+        }
+
+        companion object {
+            fun int(name: String, kotlinType: String, nullable: Boolean = false) = SqlColumn(name, "INTEGER", kotlinType, nullable)
+            fun text(name: String, kotlinType: String, nullable: Boolean = false) = SqlColumn(name, "TEXT", kotlinType, nullable)
+        }
+    }
+
+    private data class SqlTable(
+        val name: String,
+        val key: List<SqlColumn>,
+        val value: SqlColumn,
+    ) {
+        init {
+            require(key.isNotEmpty())
+            require(key.all { !it.nullable })
+        }
+
+        fun tableDefinition() = buildString {
+            append("CREATE TABLE $name (\n")
+            key.forEach { column ->
+                append("    ${column.columnDefinition()},\n")
+            }
+            append("    ${value.columnDefinition()},\n")
+            append("    PRIMARY KEY(${key.joinToString { it.name }})\n")
+            append(");\n")
+        }
+    }
+
+    private data class SqlFunction(
+        val name: String,
+        val content: StringBuilder.(SqlTable) -> Unit,
+    )
+
+    @TaskAction
+    fun run() {
+        caches.forEach { cache ->
```

Comment:
Per Kotin-style guidelines, regular for should be preferred over forEach: https://kotlinlang.org/docs/coding-conventions.html#loops

Same thing for other parts of the code.